ui/components/app/snaps/snap-ui-renderer/snap-ui-renderer.js

```diff
@@ -51,14 +51,15 @@ const SnapUIRendererComponent = ({
 
   const isValidComponent = content && isComponent(content);
 
-  const elementKeyIndex = 0;
+  const elementKeyIndex = { value: 0 };
 
   // sections are memoized to avoid useless re-renders if one of the parents element re-renders.
   const sections = useMemo(
     () =>
       isValidComponent &&
       mapToTemplate({
         element: content,
+        rootKey: nanoid(),
```

Comment:
This would mean that any time the content changes, a new root key is being generated. Is that not an issue? Just trying to understand.

Reply:
That is the idea, otherwise when the content changes (via `snap_updateInterface`) we might have duplicate keys between re-renders which causes some elements to stick around.